### ISSUE TITLE
In the diff method, empty node content and null content is being detected as a difference

### ISSUE THREAD

**Issue by rdromao** [cferdinandi/reef#20]
https://github.com/cferdinandi/reef/blob/6de9222617ba97488bb8d6e2131dce67a7869a4a/src/js/reef/reef.js#L375

If the content in the template is `null` and in the dom is `''` it would get detected as a difference and would reassign the dom node textContent when it shouldn't, right?

Maybe the validation should check that at least one of those contents is meaningful?:

```
if (templateMap[index].content !== domMap[index].content && (templateMap[index].content || domMap[index].content))
```

Btw: awesome work with this library, it allowed me to understand a lot of "state + template" things!

**Comment by cferdinandi** [comment]
@rdromao That's a good point, yea. That would happen if an element has child nodes, all of them got removed, and there was not text content inside either.

I'll make some updates...

**Comment by cferdinandi** [comment]
Fixed with v4.1.8: https://github.com/cferdinandi/reef/releases/tag/v4.1.8
